packages/lit-dev-content/site/docs/v3/components/events.md

```diff
@@ -98,6 +98,32 @@ disconnectedCallback() {
 
 See the MDN documentation on using custom elements [lifecycle callbacks](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements#Using_the_lifecycle_callbacks) for more information on `connectedCallback` and `disconnectedCallback`.
 
+### Avoiding Memory Leaks
+
+Avoid mixing and matching adding and removing event listeners between the [Lit reactive lifecycle methods](/docs/components/lifecycle/#reactive-update-cycle) and custom element [lifecycle callbacks](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements#Using_the_lifecycle_callbacks).  These two lifecycles, in some cases, operate independently of one another.  For example, just because `disconnectedCallback` has been called does not mean that `shouldUpdate`, `willUpdate`, `update`, `render`, and `updated` will not be called again if a [reactive property](/docs/components/properties/) is changed after the element is removed from the DOM.  This can result in memory leaks when listeners attached to active objects contain back references to the component that set them.
+
+```js
+connectedCallback () {
+  super.connectedCallback();
+  this.listeningForResize = false;
+}
+disconnectedCallback() {
+  window.removeEventListener("resize", this.handleWindowResize);
+  super.disconnectedCallback();
+}
+handleWindowResize = (event) => {
+  this.style.width = "880px";
+}
+willUpdate(changedProperties) {
+  if (!this.listeningForResize) {
+    window.addEventListener("resize", this.handleWindowResize);
+    this.listeningForResize = true;
+  }
+}
```

Comment:
Nit around code style – single quotes for strings and underscore for private method and unused param. Also inline comment to make it clear this shouldn't be done.
```suggestion
connectedCallback () {
  super.connectedCallback();
  this.listeningForResize = false;
}
disconnectedCallback() {
  window.removeEventListener('resize', this._handleWindowResize);
  super.disconnectedCallback();
}
_handleWindowResize = (_event) => {
  this.style.width = '880px';
}
willUpdate(_changedProperties) {
  if (!this.listeningForResize) {
    // This might never get removed and cause a memory leak!
    window.addEventListener('resize', this._handleWindowResize);
    this.listeningForResize = true;
  }
}
```